modules/greenbidsBidAdapter.js

```diff
@@ -0,0 +1,308 @@
+   * @return {Bid[]} An array of bids which were nested inside the server response.
+   */
+  interpretResponse: function (serverResponse) {
+    serverResponse = serverResponse.body;
+    if (!serverResponse.responses) {
+      return [];
+    }
+    return serverResponse.responses.map((bid) => {
+      const bidResponse = {
+        cpm: bid.cpm,
+        width: bid.width,
+        height: bid.height,
+        currency: bid.currency,
+        netRevenue: true,
+        size: bid.size,
+        ttl: bid.ttl,
+        meta: {
+          advertiserDomains: bid && bid.adomain ? bid.adomain : [],
+        },
+        ad: bid.ad,
+        requestId: bid.bidId,
+        creativeId: bid.creativeId,
+        placementId: bid.placementId,
+      };
+      if (bid.dealId) {
+        bidResponse.dealId = bid.dealId
+      }
+      if (bid?.ext?.dsa) {
+        bidResponse.meta.dsa = bid.ext.dsa;
+      }
+      return bidResponse;
+    });
+  }
+};
+
+registerBidder(spec);
+
+// Page info retrival
+
+/**
+ * Retrieves the referrer information from the bidder request.
+ *
+ * @param {Object} bidderRequest - The bidder request object.
+ * @param {Object} [bidderRequest.refererInfo] - The referer information object.
+ * @param {string} [bidderRequest.refererInfo.page] - The page URL of the referer.
+ * @returns {string} The referrer URL if available, otherwise an empty string.
+ */
+function getReferrerInfo(bidderRequest) {
+  let ref = '';
+  if (bidderRequest && bidderRequest.refererInfo && bidderRequest.refererInfo.page) {
+    ref = bidderRequest.refererInfo.page;
+  }
+  return ref;
+}
+
+/**
+ * Retrieves the title of the current web page.
+ *
+ * This function attempts to get the title from the top-level window's document.
+ * If an error occurs (e.g., due to cross-origin restrictions), it falls back to the current document.
+ * It first tries to get the title from the `og:title` meta tag, and if that is not available, it uses the document's title.
+ *
+ * @returns {string} The title of the current web page, or an empty string if no title is found.
+ */
+function getPageTitle() {
+  try {
+    const ogTitle = window.top.document.querySelector('meta[property="og:title"]');
+    return window.top.document.title || (ogTitle && ogTitle.content) || '';
+  } catch (e) {
+    const ogTitle = document.querySelector('meta[property="og:title"]');
+    return document.title || (ogTitle && ogTitle.content) || '';
+  }
+}
+
+/**
+ * Retrieves the content of the page description meta tag.
+ *
+ * This function attempts to get the description from the top-level window's document.
+ * If it fails (e.g., due to cross-origin restrictions), it falls back to the current document.
+ * It looks for meta tags with either the name "description" or the property "og:description".
+ *
+ * @returns {string} The content of the description meta tag, or an empty string if not found.
+ */
+function getPageDescription() {
```

Comment:
There's a bunch of utility functions here that can be imported, could you move them all? 

Thanks!